### PR TITLE
Add ability to access metrics from KroxyliciousTester

### PR DESCRIPTION
**Pull request by k-wall**
### Type of change

_Select the type of your PR_

- Enhancement / new feature

### Description

In #1355, we needed the ability to make observations of the metrics emitted by Kroxylicious metrics from the IT.  That PR took an inline approach, but discussions on the PR had indicated we wished to refactor.

In this PR, we augment KroxyliciousTester's API with a method `#getAdminHttpClient`, which provides an API facilitating access to Kroxylicious's admin endpoint and the metrics.   We also introduce a convenience method `#scrapeMetric` that scrapes the metric from the endpoint and provides a response in the form of `SimpleMetric` objects.

_Please describe your pull request_

### Additional Context

_Why are you making this pull request?_

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [x] Write tests
- [x] Make sure all tests pass
- [ ] Review performance test results. Ensure that any degradations to performance numbers are understood and justified.
- [x] Make sure all Sonarcloud warnings are addressed or are justifiably ignored.
- [ ] Update documentation
- [ ] Reference relevant issue(s) and close them after merging
- [ ] For user facing changes, update CHANGELOG.md (remember to include changes affecting the API of the test artefacts too).

### CODE REVIEW

File: kroxylicious-integration-test-support/src/main/java/io/kroxylicious/test/tester/SimpleMetric.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    // note: RE doesn't handle escaping within label values
+    @SuppressWarnings("java:S5852") //
+    private static final Pattern PROM_TEXT_EXPOSITION_PATTERN = Pattern
+            .compile("^(?<metric>[a-zA-Z_:][a-zA-Z0-9_:]*)(\\{(?<labels>.*)})?[\\t ]*(?<value>[0-9E.]*)[\\t ]*(?<timestamp>\\d+)?$");
+    private static final Pattern NAME_WITH_QUOTED_VALUE = Pattern.compile("^(?<name>[a-zA-Z_:][a-zA-Z0-9_:]*)=\"(?<value>.*)\"$");
+
+    static List<SimpleMetric> parse(String output) {
+        var all = new ArrayList<SimpleMetric>();
+        try (var reader = new BufferedReader(new StringReader(output))) {
+            var line = reader.readLine();
```
Comment (robobario) [inline kroxylicious-integration-test-support/src/main/java/io/kroxylicious/test/tester/SimpleMetric.java:41]: BufferedReader has a lines method that returns Stream<String>. I've pushed up a PR https://github.com/k-wall/kroxylicious/pull/17 refactoring so we don't have to work with null checks and readLine

**Review by robobario — APPROVED** [review]
LGTM

File: kroxylicious-integration-test-support/src/test/java/io/kroxylicious/test/tester/SimpleMetricTest.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+class SimpleMetricTest {
+
+    /**
+     * Known good from <a href="https://github.com/prometheus/docs/blob/main/content/docs/instrumenting/exposition_formats.md#text-format-example">exposition_formats</a>
+     */
+    static Stream<Arguments> knownGood() {
```
Comment (robobario) [inline kroxylicious-integration-test-support/src/test/java/io/kroxylicious/test/tester/SimpleMetricTest.java:26]: we should add tests for degenerate cases like empty string and comments only, I  these should be treated as valid input with an empty list as output.

**Comment by k-wall** [comment]
I hit what looks like a Sonar false positive in my work on #1405 for S5860.  I marked the issue in the Sonarcloud and reported the issue back to Sonar.

https://community.sonarsource.com/t/false-positive-for-s5860-use-the-named-groups-of-this-regex-or-remove-the-names/120384
